test/docker-e2e/e2e_block_sync_test.go

```diff
@@ -0,0 +1,201 @@
+			WithPostInit(func(ctx context.Context, node *celestiadockertypes.ChainNode) error {
+				return config.Modify(ctx, node, "config/config.toml", func(cfg *cometcfg.Config) {
+					// disable state sync to ensure we're testing block sync
+					cfg.StateSync.Enable = false
+					// configure block sync
+					cfg.BlockSync.Version = "v0"
+					// set persistent peers to connect to existing nodes
+					cfg.P2P.PersistentPeers = peerList
+				})
+			}).
+			Build(),
+	)
+
+	s.Require().NoError(err, "failed to add block sync node")
+
+	allNodes = celestia.GetNodes()
+	blockSyncNode := allNodes[len(allNodes)-1]
+
+	s.Require().Equal("fn", blockSyncNode.GetType(), "expected block sync node to be a full node")
+
+	blockSyncClient, err := blockSyncNode.GetRPCClient()
+	s.Require().NoError(err)
+
+	err = s.WaitForSync(ctx, blockSyncClient, blockSyncTimeout, func(info rpctypes.SyncInfo) bool {
+		return !info.CatchingUp && info.LatestBlockHeight >= latestHeight
+	})
+
+	s.Require().NoError(err, "failed to wait for block sync node to catch up")
+
+}
+
+// TestBlockSyncMocha tests block sync functionality by syncing from the mocha network.
+func (s *CelestiaTestSuite) TestBlockSyncMocha() {
+	t := s.T()
+	if testing.Short() {
+		t.Skip("skipping in short mode")
+	}
+
+	ctx := context.TODO()
+
+	mochaConfig := networks.NewMochaConfig()
+	mochaClient, err := networks.NewClient(mochaConfig.RPCs[0])
+	s.Require().NoError(err, "failed to create mocha RPC client")
+
+	// get latest height from mocha
+	latestHeight, err := s.GetLatestBlockHeight(ctx, mochaClient)
+	s.Require().NoError(err, "failed to get latest height from mocha")
+	s.Require().Greater(latestHeight, int64(0), "latest height is zero")
+
+	t.Logf("Mocha latest height: %d", latestHeight)
+	t.Logf("Using mocha RPC: %s", mochaConfig.RPCs[0])
+	t.Logf("Using mocha seeds: %s", mochaConfig.Seeds)
+
+	dockerCfg, err := networks.NewConfig(mochaConfig, s.client, s.network)
+	s.Require().NoError(err, "failed to create mocha config")
+
+	// create a mocha chain builder (no validators, just for block sync nodes)
+	mochaChain, err := networks.NewChainBuilder(s.T(), mochaConfig, dockerCfg).
+		WithNodes(celestiadockertypes.NewChainNodeConfigBuilder().
+			WithNodeType(celestiadockertypes.FullNodeType).
+			WithPostInit(func(ctx context.Context, node *celestiadockertypes.ChainNode) error {
+				return config.Modify(ctx, node, "config/config.toml", func(cfg *cometcfg.Config) {
+					// disable state sync to ensure we're testing block sync
+					cfg.StateSync.Enable = false
+
+					// configure block sync
+					cfg.BlockSync.Version = "v0"
+
+					// minimal p2p config like state sync test
+					cfg.P2P.Seeds = mochaConfig.Seeds
+				})
+			}).
+			Build(),
+		).
+		Build(ctx)
+
+	s.Require().NoError(err, "failed to create chain")
+
+	t.Log("Starting mocha block sync node")
+
+	// use a timeout context for starting the chain
+	startCtx, cancel := context.WithTimeout(ctx, 10*time.Minute)
```

Comment:
I noticed the full node seems to take a bit longer than usual to start, not the full 10 mins, but this should avoid any flakiness in the test.